DuckDuckGo/DuckPlayer/DuckPlayer.swift

```diff
@@ -0,0 +1,195 @@
+        case .alwaysAsk:
+            return "Ask every Time"
+        case .disabled:
+            return "Never"
+        }
+    }
+
+    init(_ duckPlayerMode: Bool?) {
+        switch duckPlayerMode {
+        case true:
+            self = .enabled
+        case false:
+            self = .disabled
+        default:
+            self = .alwaysAsk
+        }
+    }
+
+    var boolValue: Bool? {
+        switch self {
+        case .enabled:
+            return true
+        case .alwaysAsk:
+            return nil
+        case .disabled:
+            return false
+        }
+    }
+    
+    var stringValue: String {
+        switch self {
+        case .enabled:
+            return "enabled"
+        case .alwaysAsk:
+            return "alwaysAsk"
+        case .disabled:
+            return "disabled"
+        }
+    }
+
+    init?(stringValue: String) {
+        switch stringValue {
+        case "enabled":
+            self = .enabled
+        case "alwaysAsk":
+            self = .alwaysAsk
+        case "disabled":
+            self = .disabled
+        default:
+            return nil
+        }
+    }
+}
+
+/// Values that the Frontend can use to determine the current state.
+struct InitialSetupSettings: Codable {
+    struct PlayerSettings: Codable {
+        let pip: PIP
+    }
+
+    struct PIP: Codable {
+        let status: Status
+    }
+
+    enum Status: String, Codable {
+        case enabled
+        case disabled
+    }
+
+    let userValues: UserValues
+    let settings: PlayerSettings
+}
+
+/// Values that the Frontend can use to determine user settings
+public struct UserValues: Codable {
+    enum CodingKeys: String, CodingKey {
+        case duckPlayerMode = "privatePlayerMode"
+        case overlayInteracted
+    }
+    let duckPlayerMode: DuckPlayerMode
+    let overlayInteracted: Bool
+}
+
+final class DuckPlayer {
+    static let usesSimulatedRequests: Bool = {
+        if #available(macOS 12.0, *) {
+            return true
+        } else {
+            return false
+        }
+    }()
+
+    static let duckPlayerHost: String = "player"
+    static let commonName = "Duck Player"
+    
+    static let shared = DuckPlayer()
+
+    var isAvailable: Bool {
+        return true
+    }
+
+    @Published var mode: DuckPlayerMode
+
+    var overlayInteracted: Bool {
+        true
+    }
+
+    init() {
+        mode = .enabled
+    }
+
+    // MARK: - Common Message Handlers
+    
+    public func handleSetUserValuesMessage(
+        from origin: YoutubeOverlayUserScript.MessageOrigin
+    ) -> (_ params: Any, _ message: UserScriptMessage) -> Encodable? {
+
+        return { [weak self] params, _ -> Encodable? in
+            guard let self else {
+                return nil
+            }
+            guard let userValues: UserValues = DecodableHelper.decode(from: params) else {
+                assertionFailure("YoutubeOverlayUserScript: expected JSON representation of UserValues")
+                return nil
```

Comment:
Maybe add a pixel here?

Reply:
Good Call.   I added to the pixel list here: https://app.asana.com/0/1204099484721401/1207593922684294